filters/auth/oidc.go

```diff
@@ -200,27 +191,45 @@ func (f *tokenOidcFilter) validateAnyClaims(h map[string]interface{}) bool {
 		return true
 	}
 
-	var a []string
-	for k := range h {
-		a = append(a, k)
+	for i := 0; i < len(f.claims)-1; i += 2 {
+		k := f.claims[i]
+		v := f.claims[i+1]
```

Comment:
the docs might be wrong here about the claims (5th) argument of the filter: https://opensource.zalando.com/skipper/reference/filters/#oauthoidcanyclaims , but I think the docs are actually right and the claims are not key-value pairs, only a list of keys. The reason why this would make more sense to me is that we cannot know in advance the values, since without being authorized we actually shouldn't know them. I think here we should just check that at least one expected key exists from f.claims in h.

Reply:
fixed